packages/cli/src/decorators/module.ts

```diff
@@ -0,0 +1,29 @@
+import { Container, Service, type Constructable } from '@n8n/di';
+import type { ExecutionLifecycleHooks } from 'n8n-core';
+
+export interface BaseN8nModule {
+	registerLifecycleHooks?(hooks: ExecutionLifecycleHooks): void;
+}
+
+type Module = Constructable<BaseN8nModule>;
+
+export const registry = new Set<Module>();
+
+export const N8nModule = (): ClassDecorator => (target) => {
```

Comment:
This is a great start for module support 🎉 . 
But, I think all this code that's not necessarily specific to insights should be moved into a separate PR as well.